lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -129,49 +148,47 @@ def _set_config_options(self, pbm_configs):
-        try:
-            pbm_snap.start(services=["pbm-agent"])
-        except snap.SnapError as e:
-            logger.error("An exception occurred when starting pbm agent, error: %s.", str(e))
-            self.charm.unit.status = BlockedStatus("couldn't start pbm")
-            return
+        pbm_snap.start(services=["pbm-agent"])
 
-        # wait for re-sync and update charm status based on pbm syncing status.
+        # wait for re-sync and update charm status based on pbm syncing status. Need to wait for
+        # 2 seconds for pbm_agent to receive the resync command before verifying.
         subprocess.check_output("percona-backup-mongodb config --force-resync", shell=True)
-        self._verify_resync()
+        time.sleep(2)
+        self._wait_pbm_status()
 
-    def _verify_resync(self):
-        """Sets pbm status based on whether pbm can accept new configurations.
+    def _get_pbm_status(self) -> StatusBase:
+        """Retrieve pbm status."""
+        snap_cache = snap.SnapCache()
+        pbm_snap = snap_cache["percona-backup-mongodb"]
+        if not pbm_snap.present:
+            return BlockedStatus("pbm not installed.")
 
-        The status of pbm is determined by whether or not pbm_agent is able to accept its given
-        configs. Depending on whether or not pbm was able to resolve and resync config changes
-        pbm_agent will be ready/not ready for backups.
-        """
         try:
-            # get pbm status waits for resync up to 5 minutes
-            self._get_pbm_status()
-        except ResyncError as e:
-            self.charm.unit.status = WaitingStatus("waiting to sync s3 configurations.")
-            raise e
+            pbm_status = subprocess.check_output("percona-backup-mongodb status", shell=True)
+            # pbm is running resync operation
+            if "Resync" in self._current_pbm_op(pbm_status.decode("utf-8")):
+                return WaitingStatus("waiting to sync s3 configurations.")
+
+            # no operations are currently running with pbm
+            if "(none)" in self._current_pbm_op(pbm_status.decode("utf-8")):
+                return ActiveStatus("")
+
+            if "Snapshot backup" in self._current_pbm_op(pbm_status.decode("utf-8")):
+                return MaintenanceStatus("backup started/running")
+
         except subprocess.CalledProcessError as e:
             if e.returncode == CREDENTIALS_CODE:
-                logger.error("credentials for pbm are invalid.")
-                self.charm.unit.status = BlockedStatus("s3 credentials are incorrect.")
-                return
-
-            logger.error(e)
-            self.charm.unit.status = BlockedStatus("s3 configurations are incompatible.")
-            return
+                return BlockedStatus("s3 credentials are incorrect.")
 
-        self.charm.unit.status = ActiveStatus("")
+            return BlockedStatus("s3 configurations are incompatible.")
 
```

Comment:
hm, maybe, not sure
```suggestion
            return BlockedStatus("s3 configurations are incompatible.")
        return ActiveStatus("")
```

Reply:
I don't see the harm in this. AFAICT this is a solid suggestion. Thank you Mykola :)